### ISSUE TITLE
HNC: stop modifying source files whenever we call `make docker-build`

### ISSUE THREAD

**Issue by adrianludwin** [kubernetes-sigs/multi-tenancy#112]
See the following fairly terrifying lines in `Makefile` (added by `kubebuilder` when it first scaffolds the project):

```
        docker build . -t ${IMG}
        @echo "updating kustomize image patch file for manager resource"
        sed -i'' -e 's@image: .*@image: '"${IMG}"'@' ./config/default/manager_image_patch.yaml
```

This means that every time we run `make docker-build`, _our source files get modified_! We should not do this. I'm not sure what we _should_ do. Maybe have a non-default kustomization that we can `.gitignore`?

**Comment by adrianludwin** [comment]
/assign @adrianludwin 

See https://github.com/adrianludwin/multi-tenancy/commit/194bef94a9653f894f0d0c05bf42df750ca9e112

**Comment by adrianludwin** [comment]
Fixed via #161 and #164
